packages/playwright-core/src/server/registry/index.ts

```diff
@@ -1248,9 +1260,30 @@ export class Registry {
     }
   }
 
-  private async _validateInstallationCache(linksDir: string) {
-    // 1. Collect used downloads and package descriptors.
-    const usedBrowserPaths: Set<string> = new Set();
+  async list() {
+    const linksDir = path.join(registryDirectory, '.links');
+    const [browsers, _brokenLinks] = await this._traverseBrowserInstallations(linksDir);
+
+    // Group browsers by browserName
+    const groupedBrowsers: Record<string, Array<{ browserVersion: number, hostDir: string, browserPath: string }>> = {};
```

Comment:
nit: let's keep the same `BrowsersInfo` struct:
```suggestion
    const groupedBrowsers: Record<string, BrowsersInfo[]>> = {};
```